api/handler/v1beta1/grpc.go

```diff
@@ -145,7 +146,13 @@ func (s *GRPCServer) CreateProvider(ctx context.Context, req *guardianv1beta1.Cr
 		URN:    providerConfig.URN,
 		Config: providerConfig,
 	}
-	if err := s.providerService.Create(p); err != nil {
+
+	user, err := s.getUser(ctx)
+	if err != nil {
+		return nil, err
```

Comment:
any reason why we don't define error status code here and return this error directly instead?

Reply:
in case of `CreateProvider`, authenticated user email is only required for audit logging purpose, before, we're not needing it for the provider creation. And I also just realize we can log.Error it instead of returning the error then ☝️, wdyt? 

There's an example where err from `getUser` returned with the error code accordingly like in `CreateAppeal` because the user email is needed for appeal creation.

Reply:
Returning error or not (just logging it) depends on the context though, whether it is required or optional.

But in case it is optional and the worst case would be there is no user information, would it be ok for audit log to have empty 'actor'?

Reply:
yes, that's exactly the case, authorization is optional on these methods while we're getting the actor info from the authorized user email header. We are also still not sure how this should be handled.